src/app_service/service/community/service.nim

```diff
@@ -208,6 +212,18 @@ QtObject:
         var chatDetails = self.chatService.getChatById(fullChatId)
         result.add(chatDetails)
 
+  proc handleCommunitiesSettingsUpdates(self: Service, communitiesSettings: seq[CommunitySettingsDto]) =
+    for settings in communitiesSettings:
+      var emitUpdate = false
+      if self.allCommunities.hasKey(settings.id):
+        self.allCommunities[settings.id].settings = settings
+      if self.joinedCommunities.hasKey(settings.id):
```

Comment:
@PascalPrecht I've just noticed another thing here, you're checking here if `self.joinedCommunities` contains id and update `settings` prop for it and then you're emitting signal with direct access to community with `settings.id` even if it's possible that `self.joinedCommunities` doesn't contain that item. In that case we will have a crash.

Reply:
There's actually various places where we construct community items and ensure they have `settings`. So this should not be a problem in practice. Can do add a check for `settings` to safeguard, but then we might as well do that for all other properties that are expected on a community item